src/main/java/ro/utcn/sd/boti/stackoverflow/Controller/ConsoleController.java

```diff
@@ -93,6 +106,50 @@ private void listQuestions() {
     }
     private void listTags() {for (Tag t : stackoverflowService.listTags()) { print(t.toString()); } }
 
+    private Question getQuestion(){
+        print("Question ID:");
+        int id = scanner.nextInt();
+        Question question = stackoverflowService.findQuestionById(id).orElse(null);
+        if (question == null) {
+            print("Question not found!");
+            return null;
+        }
+        print(question.toString());
+        return question;
+    }
+
+    private Question getQuestionWithPermission(){
+        print("Question ID:");
+        int id = scanner.nextInt();
+        Question question = stackoverflowService.findQuestionById(id).orElse(null);
+        if (question == null) {
+            print("Question not found!");
+            return null;
+        }
+        print(question.toString());
+        if (question.getAuthor().getId() != user.getId()){
+            print("No permission");
+            return null;
+        }
+        return question;
+    }
+
+    private Answer getAnswer(){
+        print("Answer ID:");
+        int id = scanner.nextInt();
+        Answer answer = stackoverflowService.findAnswerById(id).orElse(null);
+        if (answer == null) {
+            print("Answer not found!");
+            return null;
+        }
+        print(answer.toString());
+        if (answer.getAuthor().getId() != user.getId()){
```

Comment:
This is a trace of business logic (basically it is a rule - you cannot edit answers that are not your own). As such, it should reside in the service layer.